File: backend/app/schemas/panel.py
# ============================================================
# 左侧信息面板 schema（快速待办 + 系统提醒）
# 对应参考UI左侧面板
# ============================================================
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


# ---------- 快速待办 ----------
class QuickTodoOut(BaseModel):
    id: str
    title: str
    completed: bool
    sort_order: int
    created_at: datetime
    completed_at: datetime | None = None


class CreateQuickTodoRequest(BaseModel):
    title: str = Field(min_length=1, max_length=200)


class UpdateQuickTodoRequest(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=200)
    completed: bool | None = None


# ---------- 系统提醒 ----------
class ReminderOut(BaseModel):
    id: str
    title: str
    description: str | None = None
    remind_at: datetime
    type: str
    dismissed: bool
    repeat: str
    created_at: datetime


class CreateReminderRequest(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str | None = None
    remind_at: datetime
    type: str = "custom"
    repeat: str = "none"


class UpdateReminderRequest(BaseModel):
    title: str | None = None
    description: str | None = None
    remind_at: datetime | None = None
    type: str | None = None
    dismissed: bool | None = None
    repeat: str | None = None


# ---------- 左侧面板聚合数据 ----------
class LeftPanelStats(BaseModel):
    """今日状态（5项，对齐参考UI）"""
    focus_task: int = 0       # 今日最重要
    must_do: int = 0          # 必须完成
    in_progress: int = 0      # 进行中项目
    waiting: int = 0          # 等待处理
    completed_today: int = 0  # 今日完成


class LeftPanelData(BaseModel):
    """左侧信息面板完整数据"""
    greeting: str
    date_str: str
    weekday: str
    stats: LeftPanelStats
    quick_todos: list[QuickTodoOut] = Field(default_factory=list)
    reminders: list[ReminderOut] = Field(default_factory=list)

File: frontend/src/types/panel.ts
// ============================================================
// 左侧信息面板类型（对齐后端 schemas/panel.py）
// 对应参考UI左侧：日期卡片 + 今日状态 + 待办事项 + 系统提醒
// ============================================================

/** 今日状态统计 */
export interface LeftPanelStats {
  focus_task: number
  must_do: number
  in_progress: number
  waiting: number
  completed_today: number
}

/** 快速待办 */
export interface QuickTodo {
  id: string
  title: string
  completed: boolean
  sort_order: number
  created_at: string
  completed_at: string | null
}

/** 系统提醒 */
export interface Reminder {
  id: string
  title: string
  description: string | null
  remind_at: string | null
  type: string
  dismissed: boolean
  repeat: string | null
  created_at: string
}

/** 左侧面板聚合数据 */
export interface LeftPanelData {
  greeting: string
  date_str: string
  weekday: string
  stats: LeftPanelStats
  quick_todos: QuickTodo[]
  reminders: Reminder[]
}

/** 创建快速待办 */
export interface CreateQuickTodoRequest {
  title: string
}

/** 更新快速待办 */
export interface UpdateQuickTodoRequest {
  title?: string
  completed?: boolean
}

/** 创建系统提醒 */
export interface CreateReminderRequest {
  title: string
  description?: string
  remind_at?: string
  type?: string
  repeat?: string
}

/** 更新系统提醒 */
export interface UpdateReminderRequest {
  title?: string
  description?: string
  remind_at?: string
  type?: string
  dismissed?: boolean
  repeat?: string
}
